core/controllers/admin.py

```diff
@@ -205,6 +293,54 @@ def _reload_collection(self, collection_id):
             raise Exception('Cannot reload a collection in production.')
 
 
+class AdminRoleHandler(base.BaseHandler):
+    """Handler for roles tab of admin page. Used to view and update roles."""
+
+    @require_super_admin
+    def get(self):
+        view_method = self.request.params['method']
+
+        if view_method == 'role':
+            role = self.request.params['role']
+            users_by_role = {
+                username: role
+                for username in user_services.get_usernames_by_role(role)
+            }
+            role_services.store_role_query(
+                self.user_id, feconf.VIEW_ROLE, method='role', role=role)
+            self.render_json(users_by_role)
+        elif view_method == 'username':
+            username = self.request.params['username']
+            userid = user_services.get_user_id_from_username(username)
+            role_services.store_role_query(
+                self.user_id, feconf.VIEW_ROLE, method='username',
+                username=username)
+            if userid is None:
+                self.render_json({})
+            else:
+                user_role_dict = {
+                    username: user_services.get_user_role_from_id(userid)
+                }
+                self.render_json(user_role_dict)
+        else:
+            self.render_json({'error': 'Invalid method to view'})
```

Comment:
Oh, sorry, I meant, don't we call `raise self.InvalidInputException('...')` from other controllers when the input's not valid? If you do it like this then it looks like a 200-type (i.e., successful) response.